### PR TITLE
Assign article to category

### PR DESCRIPTION
**Pull request by DwinaTech**
I added to endpoints admin/articles/add for get and post route, added select filed on admin/articles/add form and also added category schema to article schema, but I could not catch the data from category schema on article schema

### CODE REVIEW

**Comment by DwinaTech** [comment]
There are more commit I wouldn't to push them the commit I would like to review flat three commits

**Comment by DwinaTech** [comment]
I mean last three commits

**Comment by DwinaTech** [comment]
Just review it and reject it I create new PR with just story code

File: server/models/Article.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
 const mongoose = require('mongoose');
 const { Schema } = mongoose;
+const CategoryModel = require('../models/Category');
 
 const schema = new Schema({
     title: String,
-    fullContent: String
+    fullContent: String,
+    postedBy: {
```
Comment (kabaros) [inline server/models/Article.js:8]: what is postedBy?

File: server/routes/admin.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-const express = require('express');
-const router = express.Router();
+var express = require('express');
```
Comment (kabaros) [inline server/routes/admin.js:1]: why change from const to var?

File: server/routes/admin.js [hunk 3]
```diff
@@ -35,9 +35,30 @@ router.post('/categories/add', function (req, res, next) {
     res.redirect("/");
     res.end()
   }
-  categoryClient.addCategory(query, callback);
+  categoryClint.categoryClint(query, callback);
```
Comment (kabaros) [inline server/routes/admin.js:38]: `categoryClint.categoryClint` - this doesn't look right

File: server/routes/admin.js [hunk 4]
```diff
@@ -35,9 +35,30 @@ router.post('/categories/add', function (req, res, next) {
...
 });
 
+/*===========================
+        Add article
+===========================*/
+router.get('/articles/add', (req, res, next) => {
+  const callback = (error, category) => {
+    res.render('add-articles', {
+      category
```
Comment (kabaros) [inline server/routes/admin.js:47]: call the variable `categories` - plural, as it is an array of mroe than one category

File: server/views/add-articles.hbs [hunk 5]
```diff
@@ -2,14 +2,21 @@
     <h4>Title</h4>
     <input type="title" name="title" />
     <br/>
+    <h4> Categories:</h4>
+        <select>
+            <option value="null" selected>Choose a Category</option>
+            {{#each category}}
+            <option value={{ this.title }} name="categoryTitle">{{ this.title }}</option>
```
Comment (kabaros) [inline server/views/add-articles.hbs:9]: the value should be this._id I would think - you display to the user the title, but you want to save the id in the database

File: server/models/Article.js [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+        type: mongoose.Schema.Types.ObjectId,
+        ref: CategoryModel
+    },
+    categoryName: [{
```
Comment (kabaros) [inline server/models/Article.js:12]: this should be called called `categoryId` or `category` and be of type reference to CategoryModel

**Review by kabaros — CHANGES_REQUESTED** [review]
well done - you're close but still not exactly what it should do. You should expect when you save the article, that the article will be saved in the database with a category field pointing to Id of the selected category

File: server/models/Article.js (hunk 1 above)
Comment (DwinaTech) [reply server/models/Article.js:8]: to connect to category schema

File: server/routes/admin.js (hunk 2 above)
Comment (DwinaTech) [reply server/routes/admin.js:1]: just wrong I did not noticed

File: server/views/add-articles.hbs (hunk 5 above)
Comment (DwinaTech) [reply server/views/add-articles.hbs:9]: great that is true

**Pull request closed without merging**